platform/posix/transport/src/openssl_posix.c

```diff
@@ -0,0 +1,563 @@
+/* Standard includes. */
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* POSIX socket includes. */
+#include <limits.h>
+#include <errno.h>
+#include <netdb.h>
+#include <time.h>
+#include <poll.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+
+#include <sys/socket.h>
+#include <sys/types.h>
+
+/* OpenSSL include. */
+#include <openssl/ssl.h>
+
+/* Transport interface include. */
+#include "transport_interface.h"
+
+#include "openssl_posix.h"
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Timeout for transport send.
+ *
+ * @note Setting to a negative value implies an infinite timeout.
+ */
+static int tlsSendTimeout = -1;
+
+/**
+ * @brief Timeout for transport recv.
+ *
+ * @note Setting to a negative value implies an infinite timeout.
+ */
+static int tlsRecvTimeout = -1;
+
+/**
+ * @brief Definition of the HTTP network context.
+ *
+ * @note For this transport implementation, the socket descriptor and
+ * SSL context is used.
+ */
+struct NetworkContext
+{
+    int tcpSocket;
+    SSL * pSslContext;
+};
+
+/**
+ * @brief A NULL-terminated string for the SNI host name.
+ *
+ * @note If SNI is enabled, this NULL-terminated string is set to
+ * #OpensslCredentials.sniHostName, then passed to #SSL_set_tlsext_host_name.
+ */
+static char sniHostName[ HOST_NAME_MAX + 1 ];
```

Comment:
What if I need multiple instances with different values for the SNI host name?

Reply:
I am using `malloc` in order to scale to multiple instances.